core/src/main/java/org/fao/geonet/kernel/setting/SettingManager.java

```diff
@@ -457,9 +457,36 @@ String getBaseURL() {
     String getServerURL() {
         String protocol = getValue(Settings.SYSTEM_SERVER_PROTOCOL);
         String host = getValue(Settings.SYSTEM_SERVER_HOST);
-        String port = getValue(Settings.SYSTEM_SERVER_PORT);
+        Integer port = getServerPort();
 
-        return protocol + "://" + host + (isPortRequired(protocol, port) ? ":" + port : "");
+        StringBuffer sb = new StringBuffer(protocol + "://");
+
+        sb.append(host);
+
+        if (isPortRequired(protocol, port + "")) {
+            sb.append(":");
+            sb.append(port);
+        }
+
+        return sb.toString();
+    }
+
+    public Integer getServerPort() {
+        String protocol = getValue(Settings.SYSTEM_SERVER_PROTOCOL);
+
+        // some conditional logic to handle the case where there's no port in the settings
+        Integer sitePort;
+
+        Integer configuredPort = toIntOrNull(Settings.SYSTEM_SERVER_PORT);
+        if (configuredPort != null) {
+            sitePort = configuredPort;
+        } else if (protocol.equalsIgnoreCase(Geonet.HttpProtocol.HTTPS)) {
+            sitePort = Geonet.DefaultHttpPort.HTTPS;
+        } else {
+            sitePort = Geonet.DefaultHttpPort.HTTP;
+        }
```

Comment:
Starting from an empty database, I'm getting

```
Caused by: java.lang.NullPointerException
    at org.fao.geonet.kernel.setting.SettingManager.getServerPort (SettingManager.java:483)
    at org.fao.geonet.kernel.setting.SettingManager.getServerURL (SettingManager.java:460)
    at org.fao.geonet.kernel.setting.SettingManager.getBaseURL (SettingManager.java:450)
    at org.fao.geonet.api.OpenApiConfig.OpenApiConfig (OpenApiConfig.java:67)
```

Maybe we could do
```suggestion
        Integer configuredPort = getValueAsInt(Settings.SYSTEM_SERVER_PORT, -1);
        if (configuredPort != -1) {
            sitePort = configuredPort;
        } else if (protocol != null && protocol.equalsIgnoreCase(Geonet.HttpProtocol.HTTPS)) {
            sitePort = Geonet.DefaultHttpPort.HTTPS;
        } else {
            sitePort = Geonet.DefaultHttpPort.HTTP;
        }
```

And later we will have to check how to reset the OpenApiConfig when setting change.